flutter_example/lib/main.gen.dart

```diff
@@ -1,40 +1,47 @@
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
-import 'macros/auto_dispose.dart';
+import 'macros/auto_listenable.dart';
 import 'macros/functional_widget.dart';
 import 'macros/render_accessors.dart';
 
 void main() {
-  runApp(const MyApp());
+  var counterNotifier = ValueNotifier<int>(0);
+  runApp(MyApp(counterNotifier));
 }
 
 @FunctionalWidget(widgetName: 'MyApp')
-Widget _buildApp(BuildContext context,
+Widget _buildApp(BuildContext context, ValueNotifier<int> counter,
     {String? appTitle, String? homePageTitle}) {
   return MaterialApp(
       title: appTitle ?? 'Flutter Demo',
       theme: ThemeData(primarySwatch: Colors.blue),
-      home: MyHomePage(title: homePageTitle ?? 'Flutter Demo Home Page'));
+      home: MyHomePage(counter,
+          title: homePageTitle ?? 'Flutter Demo Home Page'));
 }
 
 class MyHomePage extends StatefulWidget {
-  const MyHomePage({Key? key, required this.title}) : super(key: key);
+  final ValueNotifier<int> counter;
+
+  const MyHomePage(this.counter, {Key? key, required this.title})
+      : super(key: key);
 
   final String title;
 
   @override
   State<MyHomePage> createState() => _MyHomePageState();
 }
 
+@autoListenable
 class _MyHomePageState extends State<MyHomePage> {
-  final disposable = SimpleDisposable();
-  int _counter = 0;
   Color _color = Colors.blue;
 
   void _incrementCounter() {
-    setState(() {
-      _counter++;
-    });
+    widget.counter.value++;
+  }
+
+  void _handleCounter() {
```

Comment:
It could be made explicit via an annotation on the original field (a bit weird), or maybe an annotation on this method?

It is a bit magic for sure but the macro does try to keep you on the rails (it checks for the method, and that it matches the expected signature). I think this type of convention over configuration approach is actually highly desirable for a lot of people, but we could allow other more explicit options.